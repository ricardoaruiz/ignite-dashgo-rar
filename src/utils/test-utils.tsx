import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

import { theme } from '../styles/theme'

export const renderWithTheme = (children: React.ReactNode) => {
  return render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )
}
