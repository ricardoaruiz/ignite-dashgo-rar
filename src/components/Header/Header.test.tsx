import React from 'react'

import { renderWithTheme } from '../../utils/test-utils'
import { Header } from '.'

describe('<Header />', () => {
  it('should be render header correctly', () => {
    const { container } = renderWithTheme(<Header />)

    expect(container.parentElement).toMatchSnapshot()
  })
})
