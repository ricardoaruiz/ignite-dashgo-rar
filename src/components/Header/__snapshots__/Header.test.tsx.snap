// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be render header correctly 1`] = `
<body
  class="chakra-ui-light"
>
  <div
    id="chakra-toast-portal"
  >
    <ul
      id="chakra-toast-manager-top"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-top-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-top-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-bottom-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
  <div>
    <header
      class="css-5tnzri"
    >
      <button
        aria-label="open navigation"
        class="chakra-button css-10fhv4q"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 4h18v2H3V4zm0 7h18v2H3v-2zm0 7h18v2H3v-2z"
            />
          </g>
        </svg>
      </button>
      <p
        class="chakra-text css-eytimy"
      >
        dashgo
        <span
          class="chakra-text css-134oou0"
        >
          .
        </span>
      </p>
      <div
        class="css-84nudp"
      >
        <div
          class="chakra-stack css-14g09mv"
        >
          <svg
            class="chakra-icon css-1cv0b1u"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5 18h14v-6.969C19 7.148 15.866 4 12 4s-7 3.148-7 7.031V18zm7-16c4.97 0 9 4.043 9 9.031V20H3v-8.969C3 6.043 7.03 2 12 2zM9.5 21h5a2.5 2.5 0 1 1-5 0z"
              />
            </g>
          </svg>
          <svg
            class="chakra-icon css-1cv0b1u"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M14 14.252v2.09A6 6 0 0 0 6 22l-2-.001a8 8 0 0 1 10-7.748zM12 13c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm6 6v-3h2v3h3v2h-3v3h-2v-3h-3v-2h3z"
              />
            </g>
          </svg>
        </div>
        <div
          class="css-70qvj9"
        >
          <div
            class="css-ctf7f6"
          >
            <p
              class="chakra-text css-0"
            >
              Ricardo Ruiz
            </p>
            <p
              class="chakra-text css-lw47el"
            >
              ricardo.almendro.ruiz@gmail.com
            </p>
          </div>
          <span
            class="chakra-avatar css-dzd3s1"
          >
            <div
              aria-label="Ricardo Ruiz"
              class="chakra-avatar__initials css-uo84k3"
              role="img"
            >
              RR
            </div>
          </span>
        </div>
      </div>
    </header>
  </div>
</body>
`
